vpr/src/route/route_timing.cpp

```diff
@@ -595,6 +613,16 @@ bool try_timing_driven_route_tmpl(const t_router_opts& router_opts,
 
             /* Avoid overflow for high iteration counts, even if acc_cost is big */
             pres_fac = update_pres_fac(std::min(pres_fac, static_cast<float>(HUGE_POSITIVE_FLOAT / 1e5)));
+
+            // Increase short path criticality if it's having a hard time resolving hold violations due to congestion
+            if (budgeting_inf.if_set()) {
+                bool rcv_finished = false;
+                if (itry > 5 && worst_negative_slack != 0) rcv_finished = budgeting_inf.increase_min_budgets_if_struggling(-300e-12, timing_info, worst_negative_slack, netlist_pin_lookup);
```

Comment:
`-300e-12` feels like magic.  Please move it to a `constexpr float kNamedConstant` and provide a comment for why it is the value that it is?

Reply:
That makes sense, this is a specific magic number that I would like to get rid of in the future. Right now I found that this value allows for quick hold slack resolution on most circuits, without causing minimum budgets to get too large which has other problems.